proposals/0000-componentized-cabal.rst

```diff
@@ -0,0 +1,360 @@
+.. proposal-number:: Leave blank. This will be filled in when the proposal is
+                     accepted.
+
+.. trac-ticket:: Leave blank. This will eventually be filled with the Trac
+                 ticket number which will track the progress of the
+                 implementation of the feature.
+
+.. implemented:: Leave blank. This will be filled in with the first GHC version which
+                 implements the described feature.
+
+Componentized Cabal
+===================
+
+This is a proposal for adding per-component configure to the Cabal Setup
+script API.  A prototype implementation is available at
+`#3644 <https://github.com/haskell/cabal/pull/3644>`_.
+
+Motivation
+----------
+
+A package (e.g., ``bytestring``) consists of multiple components (e.g., a
+library and its test suite). There are several motivations behind having
+multiple components in a package:
+
+* It is convenient to not have to write a second package just to hold
+  the test suite for a library. Part of this is historically motivated,
+  as in the old days it was far more difficult to do development on a
+  collection of packages than it was on a single package. But there is
```

Comment:
Also we don't support test-suite-only packages (though one could use an executable package of course).